src/psyclone/parse/kernel.py

```diff
@@ -624,29 +633,39 @@ def get_kernel_metadata(name, ast):
         raise ParseError("Kernel type {0} does not exist".format(name))
     return ktype
 
+
 def get_kernel_interface(ast):
     '''Takes the kernel module parse tree and returns the interface part
     of the parse tree.
 
     :param ast: parse tree of the kernel module code
     :type ast: :py:class:`fparser.one.block_statements.BeginSource`
 
-    :returns: Name 'name' of the parse tree of the interface 
-    :rtype: :str `fparser.one.block_statements.Interface.name`
+    :returns: Name of the parse tree of the interface block and
+              the names of the module procedures. Or None, None if
+              there is no interface or the interface has no nodule procedures.
+    :rtype: : `fparser.one.block_statements.Interface.name`,
+              `fparser.one.block_statements.Interface.a.module_procedures` '''
 
-    :returns 'None' if there is no interface, the interface has no name or has no module procedures
-    '''
     iname = None
-    for statement, _ in fpapi.walk(ast, -1):
+    sname = None
+    count = 0
+    for statement, _ in fpapi.walk(ast):
         if isinstance(statement, fparser1.block_statements.Interface):
-#           Check the interfaces assigns module procedures.
+            # count the interfaces, then can be only one!
+            count = count + 1
+            if count >= 2:
+                raise ParseError("Kernel has more than one interface.")
```

Comment:
Please make this error msg a bit more verbose (e.g. you could pass the kernel name into this routine and use it here) and "...interface; this is forbidden in the LFRic API"

Reply:
Add the kernel name as an argument to the procedure and the error message include this and is now more verbose.